lib/utils/authorizeSilently.ts

```diff
@@ -0,0 +1,52 @@
+import { SilentAuthorizationError } from "../errors";
+import { TimeoutID } from "../types";
+
+const SERVER_MESSAGE_TIMEOUT = 10000;
+
+export const authorizeSilently = (uri: string, origin: string) => {
+  return new Promise((resolve, reject) => {
+    let timeoutId: TimeoutID;
+    const iframe = document.createElement("iframe");
+    iframe.style.display = "none";
+    document.body.appendChild(iframe);
+
+    const cleanup = () => {
+      window.removeEventListener("message", onMessageHandler);
+      if (iframe && iframe.parentNode) {
+        iframe.parentNode.removeChild(iframe);
+      }
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+    };
+
+    const onMessageHandler = (event: MessageEvent) => {
+      if (event.origin !== origin) {
+        return;
+      }
+
+      cleanup();
+
+      if (!event.data.error) {
+        const { code } = event.data.response;
+        return resolve(code);
+      } else {
+        return reject(event.data.error);
+      }
+    };
+
+    window.addEventListener("message", onMessageHandler);
+    if (iframe) {
```

Comment:
Can `iframe` become `null` or `undefined` here?

Reply:
Not as far as I can tell, I added this check because at some point I had the same error as in my comment above `Object is possibly 'null'` but not anymore apparently 🤔 so I have removed the check 👍